pyiceberg/table/name_mapping.py

```diff
@@ -333,8 +334,8 @@ def struct(self, struct: StructType, struct_partner: Optional[MappedField], fiel
         return StructType(*field_results)
 
     def field(self, field: NestedField, field_partner: Optional[MappedField], field_result: IcebergType) -> IcebergType:
-        if field_partner is None:
-            raise ValueError(f"Field missing from NameMapping: {'.'.join(self.current_path)}")
+        if field_partner is None or field_partner.field_id is None:
```

Comment:
I'm curious about this change, why do we need to check for  `field_partner.field_id`

Reply:
I see `field_id` is a required field in `NestedField`. Without `field_partner.field_id is None`, type checker errors